### PR TITLE
Updated BlobContainerClient.Upload() xml documentation

### LINKED ISSUE #9470: [BUG] BlobContainerClient.UploadBlobAsync throws exception when the file already exists
**Describe the bug**
According to the documentation BlobContainerClient.UploadBlobAsync should update a blob if there is already a blob with the same name but instead it throws. 'The specified blob already exists'

***Exception or Stack Trace***
Azure.RequestFailedException: 'The specified blob already exists.'

**To Reproduce**
Just call await blobContainerClient.UploadBlobAsync("testblob.txt", stream); two times.
Or use this sample repo I uploaded:
https://github.com/ThisWillDoIt/BlobUpload 

***Code Snippet***
https://github.com/ThisWillDoIt/BlobUpload/blob/master/UpdateBlob/Program.cs

**Expected behavior**
The blob should be overwritten.

**Setup (please complete the following information):**
 - OS: Windows 10 Build 18363
 - IDE : Visual Studio 16.4.2
- .NET Core 3.1
 - Azure.Storage.Blobs 12.2.0

**Additional context**
Documentation stating in case of an existing blob it will be overwritten instead of an exception:
https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Azure.Storage.Blobs/12.2.0/api/Azure.Storage.Blobs/Azure.Storage.Blobs.BlobContainerClient.html#Azure_Storage_Blobs_BlobContainerClient_UploadBlobAsync_System_String_System_IO_Stream_System_Threading_CancellationToken_

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/9470

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs [hunk 1]
```diff
@@ -2084,10 +2084,9 @@ private void SetNameFieldsIfNull()
 
         #region UploadBlob
         /// <summary>
-        /// The <see cref="UploadBlob"/> operation creates a new block
-        /// blob or updates the content of an existing block blob in this
-        /// container.  Updating an existing block blob overwrites any existing
-        /// metadata on the blob.
+        /// The <see cref="UploadBlobAsync"/> operation creates a new block
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: Should be UploadBlob rather than UploadBlobAsync.

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: In the spirit of helping users solve the problem that started this change, should we also mention that users can call `GetBlobClient()` and then pass specialized options into `UploadAsync` if they need more fine grained control?

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs [hunk 2]
```diff
@@ -2084,10 +2084,9 @@ private void SetNameFieldsIfNull()
...
+        /// blob.  A <see cref="RequestFailedException"/> will be thrown
+        /// if the blob already exists.
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: Should we move this to the `<remarks>` section where we talk about exceptions that get thrown?

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: Moved.

File: sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs]: Fixed, and added more detail.

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**